### PR TITLE
More missilelauncher tweaks

### PR DESCRIPTION
**Pull request by StrawWagen**
Removed redundant NetworkVar

+ Will never target lfs plane seats.
+ Block switching targets when RPG is 1/4th locked on.

Move trail effect creation to client only
Was being really unreliable when ran live, Wondering if this changes it.

### CODE REVIEW

File: lua/entities/lunasflightschool_missile.lua [hunk 1]
```diff
@@ -246,10 +253,12 @@ else
...
-		local effectdata = EffectData()
-			effectdata:SetOrigin( self:GetPos() )
-			effectdata:SetEntity( self )
-		util.Effect( "lfs_missile_trail", effectdata )
+		if CLIENT then
+			local effectdata = EffectData()
+				effectdata:SetOrigin( self:GetPos() )
+				effectdata:SetEntity( self )
+			util.Effect( "lfs_missile_trail", effectdata )
+		end
```
Comment (sarahsturgeon) [inline lua/entities/lunasflightschool_missile.lua]: You can still run this serverside if you want, you just need to provide `true` to the fourth argument

```suggestion
		local effectdata = EffectData()
			effectdata:SetOrigin( self:GetPos() )
			effectdata:SetEntity( self )
		util.Effect( "lfs_missile_trail", effectdata, true, true )

```

[ref](https://wiki.facepunch.com/gmod/util.Effect)

File: lua/entities/lunasflightschool_missile.lua [hunk 2]
```diff
@@ -188,8 +188,15 @@ if SERVER then
 				effectdata:SetNormal( -self:GetForward() )
 			util.Effect( "manhacksparks", effectdata, true, true )
 
+			local dmgDone = 600
```
Comment (sarahsturgeon) [inline lua/entities/lunasflightschool_missile.lua]: nit: This variable name implies the damage has already been applied. I'd prefer something like `damageAmount` or similar

File: lua/entities/lunasflightschool_missile.lua [hunk 3]
```diff
@@ -188,8 +188,15 @@ if SERVER then
 				effectdata:SetNormal( -self:GetForward() )
 			util.Effect( "manhacksparks", effectdata, true, true )
 
+			local dmgDone = 600
+
+			if HitEnt:IsNPC() or HitEnt:IsPlayer() or HitEnt:IsNextBot() then
+				dmgDone = 100
+
+			end
```
Comment (sarahsturgeon) [inline lua/entities/lunasflightschool_missile.lua]: I don't think this belongs here - maybe the missile can have a separate method for calculating the damage on a given target. Bonus points if it has a hook

File: lua/weapons/weapon_lfsmissilelauncher.lua [hunk 4]
```diff
@@ -146,7 +150,7 @@ function SWEP:Think()
 			local stuff = WorldToLocal( vehicle:GetPos(), Angle( 0, 0, 0 ), startpos, Owner:EyeAngles() + Angle(90,0,0) )
 			local dist = stuff:Length()
 
-			if dist < ClosestDist and Ang < SmallestAng then
+			if dist < ClosestDist and Ang < SmallestAng then -- only switch when much closer!
```
Comment (sarahsturgeon) [inline lua/weapons/weapon_lfsmissilelauncher.lua]: nit: prefer comments above lines

```suggestion
			-- only switch when much closer!
			if dist < ClosestDist and Ang < SmallestAng then
```

File: lua/weapons/weapon_lfsmissilelauncher.lua [hunk 5]
```diff
@@ -155,9 +159,12 @@ function SWEP:Think()
 			end
 		end
 
-		if self:GetClosestEnt() ~= ClosestEnt then
+		print( findTime, curtime )
```
Comment (sarahsturgeon) [inline lua/weapons/weapon_lfsmissilelauncher.lua]:
```suggestion
```

Please remove before merge

File: lua/weapons/weapon_lfsmissilelauncher.lua [hunk 6]
```diff
@@ -155,9 +159,12 @@ function SWEP:Think()
 			end
 		end
 
-		if self:GetClosestEnt() ~= ClosestEnt then
+		print( findTime, curtime )
+
+		local lockingOnBlockSwitching = ( findTime + ( lockOnTime / 4 ) ) < curtime and IsValid( self:GetClosestEnt() ) and IsValid( ClosestEnt )
```
Comment (sarahsturgeon) [inline lua/weapons/weapon_lfsmissilelauncher.lua]: Not sure what this is doing, could this be its own method, or at least split into multiple calculations?

[ref](https://github.com/CFC-Servers/cfc_glua_style_guidelines#complex-expressions-should-be-written-on-multiple-lines-with-meaningful-variable-names)

File: lua/weapons/weapon_lfsmissilelauncher.lua [hunk 7]
```diff
@@ -199,7 +206,7 @@ function SWEP:PrimaryAttack()
 	self:SetNextPrimaryFire( CurTime() + 0.5 )
 	self:TakePrimaryAmmo( 1 )
 
-	timer.Simple( 0, function()
+	timer.Simple( 0.05, function()
```
Comment (sarahsturgeon) [inline lua/weapons/weapon_lfsmissilelauncher.lua]: Nonzero timers are usually a code smell. Is this the best way to handle it?

**Review by sarahsturgeon — CHANGES_REQUESTED** [review]
Just a few comments, mostly optional styling changes (I know it's not a CFC project but I mentioned them anyway)

Blocking only because of the debug print

File: lua/entities/lunasflightschool_missile.lua (hunk 3 above)
Comment (StrawWagen) [reply lua/entities/lunasflightschool_missile.lua]: Suggestion applied!

File: lua/entities/lunasflightschool_missile.lua (hunk 2 above)
Comment (StrawWagen) [reply lua/entities/lunasflightschool_missile.lua]: Applied!

File: lua/entities/lunasflightschool_missile.lua (hunk 1 above)
Comment (StrawWagen) [reply lua/entities/lunasflightschool_missile.lua]: Applied.

File: lua/weapons/weapon_lfsmissilelauncher.lua (hunk 4 above)
Comment (StrawWagen) [reply lua/weapons/weapon_lfsmissilelauncher.lua]: Applied.

File: lua/weapons/weapon_lfsmissilelauncher.lua (hunk 6 above)
Comment (StrawWagen) [reply lua/weapons/weapon_lfsmissilelauncher.lua]: Split it up.

File: lua/weapons/weapon_lfsmissilelauncher.lua (hunk 7 above)
Comment (StrawWagen) [reply lua/weapons/weapon_lfsmissilelauncher.lua]: Changed it, The reload will happen when you try to fire with an empty clip instead.

File: lua/weapons/weapon_lfsmissilelauncher.lua (hunk 5 above)
Comment (StrawWagen) [reply lua/weapons/weapon_lfsmissilelauncher.lua]: Removed it!

**Review by sarahsturgeon — APPROVED** [review]

**Pull request merged**